Made notices always visible

jorgefilipecosta opened the pull request:
## Description
This PR aims to fix the visual issue described in https://github.com/WordPress/gutenberg/issues/4128 and https://github.com/WordPress/gutenberg/issues/3777. It makes the notices always visible even when the post is scrolled.
Besides being fixed sticky behavior has required for things to look as expected in resolutions < 600px.

## How Has This Been Tested?
Add a post with much content (requires scroll).
Save the post, resize the window to width < 600px scroll from the top until the end and verify the notice is correctly shown and the sticky behavior looks as expected.
Resize the window to points between this set of number {600, 782, 960, >960}, verify that notice always adapts and is always shown in the correct position even the look changes (elements are hidden shown)

## Screenshots (jpeg or gifs if applicable):

<img width="473" alt="screen shot 2017-12-21 at 16 46 35" src="https://user-images.githubusercontent.com/11271197/34266672-fe303140-e671-11e7-8080-e4ffce910aaf.png">
<img width="476" alt="screen shot 2017-12-21 at 16 46 58" src="https://user-images.githubusercontent.com/11271197/34266673-fe4fdc98-e671-11e7-97bf-e4f45d07997c.png">
<img width="530" alt="screen shot 2017-12-21 at 16 47 19" src="https://user-images.githubusercontent.com/11271197/34266674-fe6ef402-e671-11e7-9d6e-19c03ade4878.png">
<img width="627" alt="screen shot 2017-12-21 at 16 47 40" src="https://user-images.githubusercontent.com/11271197/34266675-fe970122-e671-11e7-8d43-cf424bb28854.png">
<img width="766" alt="screen shot 2017-12-21 at 16 47 56" src="https://user-images.githubusercontent.com/11271197/34266677-feb3cbe0-e671-11e7-83b2-303f8c9ff58b.png">

cc: @afercia

jorgefilipecosta committed:
Made notices always visible; Added sticky behaviour in notices for widths lower than small breakpoint.

Used editor-left & editor-right mixins to correctly position notices.

youknowriad commented:
Mmm, this is not easy to get right because there's so many use-cases:

For example:

 - Close the Post Settings sidebar
 - Close the WP left sidebar explicitely

<img width="986" alt="screen shot 2017-12-21 at 18 34 24" src="https://user-images.githubusercontent.com/272444/34267596-c193a2a6-e67d-11e7-9a89-146ec4b170e7.png">
<img width="977" alt="screen shot 2017-12-21 at 18 33 54" src="https://user-images.githubusercontent.com/272444/34267599-c53289d6-e67d-11e7-89f6-f96fd7b4d6f2.png">


Some of these styles (those concerning the WP left sidebar) are already defined for the `.editor-header`. Might be good to extract those as a mixin. And we could do another mixin for the `right sidebar`

jorgefilipecosta committed:
Implemented editor-left&editor-right mixins.

The mixins apply editor left/right position to the selector passed as argument. It allows to use position fixed with correct dimensions.

jorgefilipecosta committed:
Used editor-left mixin for .editor-header styles.

This change removes duplicate code and solves a bug related with resizing to small screen expanding WordPress menu and then resizing back to big screen.

jorgefilipecosta commented:
Hi @youknowriad, thank you for your review. Yes, this more complex than what I expected. I refactored this and created a mixin as you suggested. The mixin uses the logic implemented for .editor-header, but I found an existing bug and I added minor changes to the logic.
The bug can be replicated by resizing the screen to mobile, opening the responsive menu, keeping it open and resize back to the desktop. The width keeps the same as when the responsive menu was opened no matter if the menu is collapsed or not.
Before:
<img width="747" alt="screen shot 2017-12-22 at 14 50 04" src="https://user-images.githubusercontent.com/11271197/34302364-68f0a800-e728-11e7-9c27-4ca8a7676cc0.png">

After:
<img width="1115" alt="screen shot 2017-12-22 at 14 48 40" src="https://user-images.githubusercontent.com/11271197/34302371-6fe13fa8-e728-11e7-9d43-c02f6b9d299c.png">

youknowriad commented on editor/assets/stylesheets/_mixins.scss:
```diff
@@ -143,3 +143,60 @@ $float-margin: calc( 50% - #{ $visual-editor-max-width-padding / 2 } );
+/**
+ * Applies editor left position to the selector passed as argument
+ */
+@mixin editor-left( $selector ) {
```
Is it possible to avoid the selector parameter and use `&`? that way, we could embed the `@include` call instead of passing the selector as a parameter. I don't know if it's possible personally but seems nicer if so.

jorgefilipecosta replied:
Hi @youknowriad, you mean use something like:
.components-notice-list {
@include editor-left;
}
Unfortunately, I don't think this is possible in our case. Even if we could use & all the styles included would be applied to a descendant of .components-notice-list

youknowriad commented:
I really like these mixins 👍

youknowriad reviewed (approved):
Nice work here
